src/offset_date_time.rs

```diff
@@ -1320,4 +1332,29 @@ impl From<OffsetDateTime> for SystemTime {
         }
     }
 }
+
+#[allow(clippy::fallible_impl_from)]
+#[cfg(all(
+    target_arch = "wasm32",
+    not(any(target_os = "emscripten", target_os = "wasi"))
+))]
+impl From<js_sys::Date> for OffsetDateTime {
+    fn from(js_date: js_sys::Date) -> Self {
+        let timestamp = (js_date.get_time() / 1000.0) as i64;
+        Self::from_unix_timestamp(timestamp)
+            .expect("invalid timestamp: Timestamp cannot fit in range")
+    }
+}
```

Comment:
The clippy lint exists for a reason — extra care is needed. As written, this function would only work for 292,000 years, while this crate with the `large-dates` feature enabled supports ±999,999. Further, the case to `i64` discards information.

What needs to occur is for this is to construct the value using `from_unix_timestamp_nanos`. The value will need to be multiplied by 1,000,000 first, then case to `i128`. _Then_ you can safely `.expect`, as the range of integers is well outside the supported range of values.

Reply:
Needs to be a million, not a thousand.

Reply:
In js both the [Date constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date#time_value_or_timestamp_number) and [getTime()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getTime) take and return the value in milliseconds since the Unix epoch:
```
> new Date(1)
1970-01-01T00:00:00.001Z
> new Date(1000)
1970-01-01T00:00:01.000Z
```

hence why I put a thousand in both.

Reply:
Yes, but we need nanoseconds. That's why it should be multiplied by a million, rather than the current code which is multiplied by a thousand.

Reply:
My bad I can't count x) thought 1 millisecond was 1000 nanoseconds, changed